src/index.js

```diff
@@ -0,0 +1,37 @@
+odometer = function(array) {
+  let addOne = true
+
+  for (let i = array.length; i--; i > -1) {
+    let digit = array[i]
+
+    if (digit === 9 && addOne) {
+      array[i] = 0
+      addOne = true
+    } else if (addOne) {
+      array[i] += 1
+      addOne = false
+    }
+  }
+
+  return array
+}
+
+odometerReverse = function(array) {
+  let minusOne = true
+
+  for (let i = array.length; i--; i > -1) {
```

Comment:
Interesting... this clearly works, but doesn't look like a normal `for` loop. Can you explain why it works as-is? How else might you write this?

Reply:
Oh, very interesting indeed... I believe what is happening is when the loop runs, the second statement, `i--`, is decrementing. Normally this would happen in the third statement but that always returning `true`. However the "should loop continue" statement is getting to 0 and terminating the loop. I believe this is demonstrated by this repl https://repl.it/@dustinyschild/CrushingStripedWorker

Reply:
This loop should be rewritten as follows:
```
for (let i = array.length - 1; i === 0; i--)
```

Reply:
Are you sure?

Reply:
Nope... But this is correct. Should have ran tests first!
`for (let i = array.length - 1; i > -1; i--)`